contracts/DomainRegistar.sol

```diff
@@ -3,21 +3,44 @@ pragma solidity ^0.8.25;
 
 // Uncomment this line to use console.log
 // import "hardhat/console.sol";
-import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
-import  {DomainUtils} from "./DomainUtils.sol";
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { AggregatorV3Interface } from "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol";
+import { DomainUtils } from "./DomainUtils.sol";
 
+/**
+ * Emitted on domain registration
+ * @param _owner domain owner
+ * @param owner domain owner
+ * @param domain name of a new domain 
+ * @param subdomainPriceUsdc USDC price for subdomain registration
+ */
+event DomainRegistered(address indexed _owner, address owner, string domain, uint256 subdomainPriceUsdc);
+
+/**
+ * Emitted on price change
+ * @param newPrice new USDC price
+ * @param oldPrice old USDC price
+ */
+event PriceChanged(uint256 newPrice, uint256 oldPrice);
+
+/// Raised when a caller is not allowed to call the function
 error AccessDenied(string description);
+/// Raised when coins provided for domain registration is lower that subdomain price
 error NotEnoughFunds(uint256 provided, uint256 required);
+/// Raised when the requested domain name already exists
 error DuplicateDomain();
+/// Raised when domain name includes extra symbols (/*7&^ etc)
 error InvalidDomainName();
+/// Raised on attempt to create a subdomain under missing parent domain, like requested.missingparent
 error ParentDomainDoesNotExists();
-
+/// Raised on error during earnings withdrawal
+error WithdrawalFailure();
 
 /**
  * @title A simple contract for top-level domain registration
  * @author Me
  */
-contract DomainRegistar is Initializable {
+contract DomainRegistar {
```

Comment:
is this contract stay upgradable?

Reply:
it is not. Preserving upgrade functionality was not required for this task. The main objective is inter-contract communication (DomainRegistar, USDC token, Datafeed) and using a contract outside by frontend/backend